torchx/schedulers/kubernetes_mcad_scheduler.py

```diff
@@ -0,0 +1,1121 @@
+        node_selector[LABEL_INSTANCE_TYPE] = resource.capabilities[LABEL_INSTANCE_TYPE]
+
+    # To support PyTorch dataloaders we need to set /dev/shm to larger than the
+    # 64M default so we mount an unlimited sized tmpfs directory on it.
+    SHM_VOL = "dshm"
+    volumes = [
+        V1Volume(
+            name=SHM_VOL,
+            empty_dir=V1EmptyDirVolumeSource(
+                medium="Memory",
+            ),
+        ),
+    ]
+    volume_mounts = [
+        V1VolumeMount(name=SHM_VOL, mount_path="/dev/shm"),
+    ]
+    security_context = V1SecurityContext()
+
+    for i, mount in enumerate(role.mounts):
+        mount_name = f"mount-{i}"
+        if isinstance(mount, BindMount):
+            volumes.append(
+                V1Volume(
+                    name=mount_name,
+                    host_path=V1HostPathVolumeSource(
+                        path=mount.src_path,
+                    ),
+                )
+            )
+            volume_mounts.append(
+                V1VolumeMount(
+                    name=mount_name,
+                    mount_path=mount.dst_path,
+                    read_only=mount.read_only,
+                )
+            )
+        elif isinstance(mount, VolumeMount):
+            volumes.append(
+                V1Volume(
+                    name=mount_name,
+                    persistent_volume_claim=V1PersistentVolumeClaimVolumeSource(
+                        claim_name=mount.src,
+                    ),
+                )
+            )
+            volume_mounts.append(
+                V1VolumeMount(
+                    name=mount_name,
+                    mount_path=mount.dst_path,
+                    read_only=mount.read_only,
+                )
+            )
+        elif isinstance(mount, DeviceMount):
+            volumes.append(
+                V1Volume(
+                    name=mount_name,
+                    host_path=V1HostPathVolumeSource(
+                        path=mount.src_path,
+                    ),
+                )
+            )
+            volume_mounts.append(
+                V1VolumeMount(
+                    name=mount_name,
+                    mount_path=mount.dst_path,
+                    read_only=(
+                        "w" not in mount.permissions and "m" not in mount.permissions
+                    ),
+                )
+            )
+            security_context.privileged = True
+        else:
+            raise TypeError(f"unknown mount type {mount}")
+
+    torchx_env_var = [
+        V1EnvVar(
+            name=name,
+            value=value,
+        )
+        for name, value in role.env.items()
+    ]
+
+    my_env_var = [
+        V1EnvVar(
+            name=f"MCAD_{cleanup_str(role.name)}_{replica_id}_HOSTS".upper().replace(
+                "-", ""
+            ),
+            value=f"{unique_app_id}-{replica_id}.{unique_app_id}",
+        )
+        for replica_id in range(role.num_replicas)
```

Comment:
nit: these environment variables aren't strictly needed, just one for the rank 0 host

may want to prefix it with TORCHX_MCAD so users don't think they're provided by the scheduler

Reply:
Updated prefix in commit: bcb0ebe432db59f8ecca3c0ac77bd8bdc2d5a83a